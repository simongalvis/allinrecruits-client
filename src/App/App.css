@import url('https://fonts.googleapis.com/css2?family=Oswald:wght@300&display=swap');

.App {
  text-align: center;
  height:100%;
}
body{
  font-family: 'Oswald', sans-serif;
}
