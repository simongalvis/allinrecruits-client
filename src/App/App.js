import React from 'react';
import { Route } from 'react-router-dom';
import './App.css';
import LandingPage from '../LandingPage/LandingPage'
import Footer from '../Footer/Footer';
import ApplicationPage from '../ApplicationPage/ApplicationPage';
import AdminLogin from '../AdminLogin/AdminLogin';
import AdminDashboard from '../AdminDashboard/AdminDashboard';
import ApplicantList from '../ApplicantList/ApplicantList'



class App extends React.Component {






renderMainRoutes(){
  return(
  <>
    <Route exact path="/" component={LandingPage} />
    <Route exact path="/application-page" component={ApplicationPage}/>
    <Route exact path="/admin-login" component={AdminLogin}/>
    <Route exact path="/admin-dashboard" component={AdminDashboard}/>
    <Route exact path="/applicant-list" component={ApplicantList}/>

  </>
  )
  
}
  render(){
    return (
      <div className="App">
       <main>
         {this.renderMainRoutes()}
       </main>
       <Footer/>
      </div>
    );

  }
 
}

export default App;
