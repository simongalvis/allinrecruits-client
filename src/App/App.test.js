import React from 'react'
import ReactDOM from 'react-dom'
import App from './App'
import renderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'

describe(`App component`, () =>{
    it(`renders without crashing`, () =>{
        const div = document.createElement('div');

        ReactDOM.render(<BrowserRouter><App/></BrowserRouter>, div)

        ReactDOM.unmountComponentAtNode(div);
    })
    it(`Renders UI as expected`, () =>{
        const tree = renderer
            .create(<BrowserRouter><App/></BrowserRouter>)
            .toJSON();
        expect(tree).toMatchSnapshot();
    })
})