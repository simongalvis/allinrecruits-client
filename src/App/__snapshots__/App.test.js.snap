// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App component Renders UI as expected 1`] = `
<div
  className="App"
>
  <main>
    <div
      className="LandingPage"
    >
      <div
        className="NavBar"
      >
        <header>
          <a
            href="/"
            onClick={[Function]}
          >
            <img
              alt="all-in-education-logo"
              id="main-logo"
              src="all-in-logo.png"
            />
          </a>
          <a
            href="/admin-login"
          >
            <button
              id="admin-login-button"
            >
              Admin Login
            </button>
          </a>
        </header>
      </div>
      <h1>
        Welcome to All-In Recruits!
      </h1>
      <p
        id="hero-text"
      >
        Find a position as a: 
         
          
        <br />
        <div
          className="typingHero"
        >
          <b
            className="styles_typicalWrapper__1_Uvh"
          />
        </div>
      </p>
      <br />
      <a
        href="/application-page"
        onClick={[Function]}
      >
        <button
          id="apply-button"
        >
          Apply Now
        </button>
      </a>
    </div>
  </main>
  <div
    className="Footer"
  >
    <p
      className="portfolio-link"
    >
      Â© - 
      <a
        className="portfolio-link"
        href="http://simongalvis.com/"
        rel="noopener noreferrer"
        target="_blank"
      >
        simongalvis.com
      </a>
    </p>
  </div>
</div>
`;
